java/ql/src/semmle/code/xml/MavenPom.qll

```diff
@@ -365,6 +365,17 @@ class PomProperty extends PomElement {
   PomProperty() { getParent() instanceof PomProperties }
 }
 
+/**
+ * An XML element representing any kind of repository declared inside of a Maven POM XML file.
+ */
+class DeclaredRepository extends PomElement {
+  DeclaredRepository() {
+    this.getName() = ["repository", "snapshotRepository", "pluginRepository"]
+  }
+
+  string getUrl() { result = getAChild("url").(PomElement).getValue() }
```

Comment:
Needs docstring

Reply:
Other getters in this file don't have docstrings. Does this really need one?

Reply:
The rule is if public then has docstring; this is enforced by various CI and check scripts, so it's easiest to just give it a simple docstring

Reply:
There's a CI check that requires that anything new adheres to this rule, so it won't pass CI without it. We still haven't quite caught up everywhere, which is why there's still some existing code that doesn't follow this rule yet.

Reply:
Done!